showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,554 @@
+import java.util.concurrent.TimeUnit;
+import org.junit.AfterClass;
+import org.junit.Test;
+
+public class ITOtelMetrics {
+
+  @AfterClass
+  public static void cleanup_otelcol() throws Exception {
+    Process process = Runtime.getRuntime().exec("../scripts/cleanup_otelcol.sh");
+    process.waitFor();
+  }
+
+  @Test
+  public void testHttpJson_OperationSucceded_recordsMetrics() throws Exception {
+
+    // initialize the otel-collector
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory1",
+        "../opentelemetry-helper/configs/testHttpJson_OperationSucceeded.yaml");
+
+    EchoSettings httpJsonEchoSettings =
+        EchoSettings.newHttpJsonBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTracerFactory(createOpenTelemetryTracerFactory("4317"))
+            .setTransportChannelProvider(
+                EchoSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+
+    EchoClient client = EchoClient.create(httpJsonEchoSettings);
+
+    EchoRequest requestWithNoError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.OK.ordinal()).build())
+            .build();
+
+    client.echoCallable().futureCall(requestWithNoError).isDone();
+
+    // wait for the metrics to get uploaded
+    Thread.sleep(5000);
+
+    String filePath = "../directory1/testHttpJson_OperationSucceeded_metrics.txt";
+    // 1. Check if the log file exists
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    client.close();
+    client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testGrpc_OperationSucceded_recordsMetrics() throws Exception {
+
+    // initialize the otel-collector
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory2",
+        "../opentelemetry-helper/configs/testGrpc_OperationSucceeded.yaml");
+
+    EchoSettings grpcEchoSettings =
+        EchoSettings.newBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTracerFactory(createOpenTelemetryTracerFactory("4318"))
+            .setTransportChannelProvider(
+                EchoSettings.defaultGrpcTransportProviderBuilder()
+                    .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+                    .build())
+            .setEndpoint("localhost:7469")
+            .build();
+
+    EchoClient client = EchoClient.create(grpcEchoSettings);
+
+    EchoRequest requestWithNoError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.OK.ordinal()).build())
```

Comment:
I think you can do `setContent(...)` for a successful response for the echo RPC.

Reply:
Done.